nucypher/cli/commands/worklock.py

```diff
@@ -378,15 +379,25 @@ def enable_claiming(general_config, worklock_options, force, hw_wallet, gas_limi
         emitter.echo(BIDS_VALID_NO_FORCE_REFUND_INDICATED, color='yellow')
 
     if not bidder.worklock_agent.bidders_checked():
-        if not gas_limit:
-            # TODO print gas estimations
-            min_gas = 180000
-            gas_limit = click.prompt(PROMPT_BID_VERIFY_GAS_LIMIT.format(min_gas=min_gas), type=click.IntRange(min=min_gas))
 
-        if not force:
-            confirmation = CONFIRM_BID_VERIFICATION.format(bidder_address=worklock_options.bidder_address,
-                                                           gas_limit=gas_limit)
-            click.confirm(confirmation, abort=True)
+        confirmation = gas_limit and force
+        while not confirmation:
+            if not gas_limit:
+                min_gas = 180000
+                gas_limit = click.prompt(PROMPT_BID_VERIFY_GAS_LIMIT.format(min_gas=min_gas), type=click.IntRange(min=min_gas))
+
+            bidders_per_transaction = bidder.worklock_agent.estimate_verifying_correctness(gas_limit=gas_limit)
+
+            if not force:
+                message = CONFIRM_BID_VERIFICATION.format(bidder_address=worklock_options.bidder_address,
+                                                          gas_limit=gas_limit,
+                                                          bidders_per_transaction=bidders_per_transaction)
+                confirmation = click.confirm(message)
```

Comment:
Do we no longer want to abort if confirmation fails?

Reply:
only here, to make possible try different gas amount before confirm, also to abort can use ctrl+C or similar

Reply:
👍 Oh right, the while loop.